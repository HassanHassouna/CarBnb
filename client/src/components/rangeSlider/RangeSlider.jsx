import { useState } from 'react';
import Box from '@mui/material/Box';
import Slider from '@mui/material/Slider';
import { Typography } from '@mui/material';

function valuetext(value) {
  return `${value}$`;
}

export default function RangeSlider({
  minValue = 0,
  maxValue = 100,
  onChangeRange = () => {},
}) {
  const [value, setValue] = useState([minValue, maxValue]);

  const handleChange = (event, newValue) => {
    onChangeRange(value);
    setValue(newValue);
  };

  return (
    <Box sx={{ width: '90%' }}>
      <Typography variant={'subtitle1'}>
        ${value[0]}-{value[1]}/day{' '}
      </Typography>
      <Slider
        getAriaLabel={() => 'price range'}
        value={value}
        onChange={handleChange}
        valueLabelDisplay="auto"
        getAriaValueText={valuetext}
        max={maxValue}
        min={minValue}
      />
    </Box>
  );
}
