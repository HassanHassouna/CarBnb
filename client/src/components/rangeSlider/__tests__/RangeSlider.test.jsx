import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import RangeSlider from '../RangeSlider';

describe('RangeSlider', () => {
  test('should render with no parameters', () => {
    const { slider } = render(<RangeSlider />);

    expect(slider).toMatchSnapshot();
  });
});
