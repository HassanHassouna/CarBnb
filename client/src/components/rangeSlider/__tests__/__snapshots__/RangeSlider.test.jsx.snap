// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeSlider should render with no parameters 1`] = `undefined`;
